rust-src/concordium_base/src/protocol_level_tokens/token_event.rs

```diff
@@ -99,20 +106,27 @@ pub struct TokenListUpdateEventDetails {
 }
 
 /// An event emitted when the token is paused or unpaused.
-#[derive(
-    Debug,
-    Clone,
-    Eq,
-    PartialEq,
-    serde::Serialize,
-    serde::Deserialize,
-    CborSerialize,
-    CborDeserialize,
-)]
+#[derive(Debug, Clone, Eq, PartialEq, serde::Serialize, serde::Deserialize)]
 #[serde(rename_all = "camelCase")]
-pub struct TokenPauseEventDetails {
-    /// Whether the token is paused or not.
-    pub paused: bool,
+pub struct TokenPauseEventDetails {}
+
+impl CborSerialize for TokenPauseEventDetails {
+    fn serialize<C: cbor::CborEncoder>(&self, encoder: C) -> CborSerializationResult<()> {
+        encoder.encode_map(0)?;
+        Ok(())
+    }
+}
+
+impl CborDeserialize for TokenPauseEventDetails {
+    fn deserialize<C: cbor::CborDecoder>(decoder: C) -> CborSerializationResult<Self>
+    where
+        Self: Sized, {
+        let map_decoder = decoder.decode_map()?;
+        if map_decoder.size() != 0 {
+            return Err(CborSerializationError::map_size(0, map_decoder.size()));
+        }
+        Ok(TokenPauseEventDetails {})
+    }
 }
```

Comment:
@allanbrondum ideally I would have used the derive macros for this, but no matter what I did I could not get it to work with empty structs. I tried to go look at the macro code, but that's all a little foreign for me so I gave up.

If it's an easy fix for you, you're welcome to submit a PR to change this 😄 I think it would be good to be able to handle this.